pkg/steps/cluster_claim.go

```diff
@@ -99,71 +125,59 @@ func acquireCluster(ctx context.Context, clusterClaim api.ClusterClaim, hiveClie
 	}
+	ret = claim
 	logrus.Info("The claimed cluster is ready.")
 	clusterDeploymentName := claim.Spec.Namespace
 	clusterDeploymentNamespace := claim.Spec.Namespace
 	clusterDeployment := &hivev1.ClusterDeployment{}
-	if err := hiveClient.Get(ctx, ctrlruntimeclient.ObjectKey{Name: clusterDeploymentName, Namespace: clusterDeploymentNamespace}, clusterDeployment); err != nil {
-		return claim, fmt.Errorf("failed to get cluster deployment %s in namespace %s: %w", clusterDeploymentName, clusterDeploymentNamespace, err)
+	if err := s.hiveClient.Get(ctx, ctrlruntimeclient.ObjectKey{Name: clusterDeploymentName, Namespace: clusterDeploymentNamespace}, clusterDeployment); err != nil {
+		return ret, fmt.Errorf("failed to get cluster deployment %s in namespace %s: %w", clusterDeploymentName, clusterDeploymentNamespace, err)
 	}
 	if clusterDeployment.Spec.ClusterMetadata == nil {
-		return claim, fmt.Errorf("got nil cluster metadata from cluster deployment %s in namespace %s", clusterDeploymentName, clusterDeploymentNamespace)
+		return ret, fmt.Errorf("got nil cluster metadata from cluster deployment %s in namespace %s", clusterDeploymentName, clusterDeploymentNamespace)
 	}
 	kubeconfigSecretName := clusterDeployment.Spec.ClusterMetadata.AdminKubeconfigSecretRef.Name
 	passwordSecretName := clusterDeployment.Spec.ClusterMetadata.AdminPasswordSecretRef.Name
 
 	for src, dst := range map[string]string{kubeconfigSecretName: api.HiveAdminKubeconfigSecret, passwordSecretName: api.HiveAdminPasswordSecret} {
 		srcSecret := &corev1.Secret{}
-		if err := hiveClient.Get(ctx, ctrlruntimeclient.ObjectKey{Name: src, Namespace: clusterDeploymentNamespace}, srcSecret); err != nil {
-			return claim, fmt.Errorf("failed to get secret %s in namespace %s: %w", kubeconfigSecretName, clusterDeploymentNamespace, err)
+		if err := s.hiveClient.Get(ctx, ctrlruntimeclient.ObjectKey{Name: src, Namespace: clusterDeploymentNamespace}, srcSecret); err != nil {
+			return ret, fmt.Errorf("failed to get secret %s in namespace %s: %w", kubeconfigSecretName, clusterDeploymentNamespace, err)
 		}
-		dstSecret, err := mutate(srcSecret, dst, jobSpec.Namespace())
+		dstNS := s.jobSpec.Namespace()
+		dstSecret, err := mutate(srcSecret, dst, dstNS)
 		if err != nil {
-			return claim, fmt.Errorf("failed to mutate secret: %w", err)
+			return ret, fmt.Errorf("failed to mutate secret: %w", err)
 		}
-		if _, err := util.UpsertImmutableSecret(ctx, client, dstSecret); err != nil {
-			return claim, fmt.Errorf("failed to upsert immutable secret %s in namespace %s: %w", dstSecret.Name, dstSecret.Namespace, err)
+		if _, err := util.UpsertImmutableSecret(ctx, s.client, dstSecret); err != nil {
+			return ret, fmt.Errorf("failed to upsert immutable secret %s in namespace %s: %w", dst, dstNS, err)
 		}
 	}
-	return claim, nil
+	return ret, nil
 }
 
 func mutate(secret *corev1.Secret, name, namespace string) (*corev1.Secret, error) {
```

Comment:
nit: if you pass a pointer into a function called `mutate` you expect the underlying object to be mutated, not returned

Reply:
also: `mutate` is a poor function name, since it does not describe whatsoever _what_ is going on or _why_

Reply:
Renamed.